Dockerfile

```diff
@@ -13,8 +13,6 @@
 #
 # To build the image from a local source checkout:
 # $ docker build -t splinterdb .
-# To then run the splinterdb-cli within the build image:
-# $ docker run --rm splinterdb bin/splinterdb-cli --help
```

Comment:
Why is this going away?

Reply:
@ajhconway - Gabriel pulled-out support for Rust-based CLI tooling under PR #307 @ SHA 916625a.

This comment in the `Dockerfile` is a straggling reference.